gateway/src/main/java/org/sample/payment/gateway/service/ValidationImpl.java

```diff
@@ -0,0 +1,232 @@
+package org.sample.payment.gateway.service;
+
+import org.apache.commons.validator.routines.checkdigit.LuhnCheckDigit;
+
+import org.sample.payment.gateway.exception.CustomValidationException;
+import org.sample.payment.gateway.model.Card;
+import org.sample.payment.gateway.model.Cardholder;
+import org.sample.payment.gateway.model.Payment;
+import org.sample.payment.gateway.repository.PaymentRepository;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+import javax.xml.bind.ValidationException;
+
+import java.time.LocalDate;
+import java.util.HashMap;
+import java.util.Map;
+
+import static org.sample.payment.gateway.constants.Constant.*;
+
+@Service
+public class ValidationImpl implements Validation{
+
+//    private ValidationImpl validation;
+
+
+    public PaymentRecordService paymentRecordService;
+
+    @Autowired
+    public ValidationImpl(PaymentRecordService paymentRecordService) {
+        this.paymentRecordService = paymentRecordService;
+    }
+
+    private Map<String, String> errors;
+
+    @Override
+    public Map<String, String> validatePayment(Payment payment) throws CustomValidationException {
+        errors = new HashMap<>();
+        try {
+            validateInvoiceField(payment.getInvoice());
+        } catch (CustomValidationException exception) {
```

Comment:
Fill-in errors map directly without try-catch. Throw exception inly at the end when map is not empty